platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -184,13 +161,20 @@ public void serialize(
             @NonNull final SerializableDataOutputStream out, @NonNull final EventSerializationOptions option)
             throws IOException {
         out.writeSerializable(softwareVersion, true);
-        // FUTURE WORK: The creatorId should be a selfSerializable NodeId at some point.
-        // Changing the event format may require a HIP.  The old format is preserved for now.
-        out.writeLong(creatorId.id());
-        out.writeLong(selfParentGen);
-        out.writeLong(otherParentGen);
-        out.writeSerializable(selfParentHash, false);
-        out.writeSerializable(otherParentHash, false);
+        if (deserializedVersion < ClassVersion.ADDRESS_BOOK_ROUND) {
+            // FUTURE WORK: The creatorId should be a selfSerializable NodeId at some point.
+            // Changing the event format may require a HIP.  The old format is preserved for now.
+            out.writeLong(creatorId.id());
```

Comment:
can we make this change now? so that `NodeId` is serialized?

Reply:
I should have deleted the comment which is in the section of the code for the older serialization scheme.   The body of the else does serialize the node id as a SelfSerializable.